protocols/migration/migration_reports_4_0_0_to_reports_4_2_0.py

```diff
@@ -0,0 +1,237 @@
+            structural_tiered_variants.append(self.migrate_reported_structural_variant_cancer(
+                reported_structural_variant
+            ))
+        new_cancer_interpretation_request.structuralTieredVariants = structural_tiered_variants
+
+        return self.validate_object(
+            object_to_validate=new_cancer_interpretation_request, object_type=self.new_model.CancerInterpretationRequest
+        )
+
+    def migrate_cancer_interpreted_genome(self, cancer_interpreted_genome, sample_id):
+        """
+        :type cancer_interpreted_genome: reports_4_0_0.CancerInterpretedGenome
+        :type sample_id: str
+        :rtype: reports_4_2_0.CancerInterpretedGenome
+        """
+        new_cancer_interpreted_genome = self.new_model.CancerInterpretedGenome.fromJsonDict(
+            jsonDict=cancer_interpreted_genome.toJsonDict()
+        )
+
+        new_cancer_interpreted_genome.gitVersionControl = "4.2.0"
+
+        # FIXME: we are using the individual identifier in the cancer participant to set the sample id
+        reported_variants = []
+        for reported_variant in cancer_interpreted_genome.reportedVariants:
+            reported_variants.append(self.migrate_reported_variant_cancer(
+                reported_variant,
+                sample_id
+            ))
+        new_cancer_interpreted_genome.reportedVariants = reported_variants
+
+        # FIXME: we have no sample id field in the structural variant
+        structural_reported_variants = []
+        for reported_structural_variant in cancer_interpreted_genome.reportedStructuralVariants:
+            structural_reported_variants.append(self.migrate_reported_structural_variant_cancer(reported_structural_variant))
+        new_cancer_interpreted_genome.reportedStructuralVariants = structural_reported_variants
+
+        return self.validate_object(
+            object_to_validate=new_cancer_interpreted_genome, object_type=self.new_model.CancerInterpretedGenome
+        )
+
+    def migrate_cancer_clinical_report(self, cancer_clinical_report, sample_id):
+        """
+        :type cancer_clinical_report: reports_4_0_0.ClinicalReportCancer
+        :type sample_id: str
+        :rtype: reports_4_2_0.ClinicalReportCancer
+        """
+        new_cancer_clinical_report = self.new_model.ClinicalReportCancer.fromJsonDict(
+            jsonDict=cancer_clinical_report.toJsonDict()
+        )
+
+        # FIXME: we are using the individual identifier in the cancer participant to set the sample id
+        candidate_variants = []
+        for candidate_variant in cancer_clinical_report.candidateVariants:
+            candidate_variants.append(self.migrate_reported_variant_cancer(
+                candidate_variant,
+                sample_id
+            ))
+        new_cancer_clinical_report.reportedVariants = candidate_variants
+
+        # FIXME: we have no sample id field in the structural variant
```

Comment:
@priesgo do we need to change 4.2.0 before closing it ?

Reply:
no!!